phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/PagingRegionScanner.java

```diff
@@ -59,8 +70,32 @@ public PagingRegionScanner(Region region, RegionScanner scanner, Scan scan) {
         if (pagingFilter != null) {
             pagingFilter.init();
         }
+
+        TableDescriptor tableDescriptor = region.getTableDescriptor();
+        BloomType bloomFilterType = tableDescriptor.getColumnFamilies()[0].getBloomFilterType();
+        if (bloomFilterType == BloomType.ROW) {
+            // Check if the scan is a multi-point-lookup scan
+            skipScanFilter = ScanUtil.getSkipScanFilter(scan);
+            if (skipScanFilter != null && skipScanFilter.isMultiKeyPointLookup()) {
+                isMultiKeyPointLookup = true;
+                multiKeyPointLookupRanges = skipScanFilter.getPointLookupKeyRanges();
+            }
+        }
     }
 
+    private void setMultiKeyPointLookupPosition(byte[] rowKey) throws IOException {
```

Comment:
Since this is meant to be used when high level scanners need to re-scan the already scanned rows, this is kind of resetting `multiKeyPointLookupPosition` to 0 and then iterating through key ranges. So, shall we call this `resetMultiKeyPointLookupPosition` or `setMultiKeyPointLookupPositionToAdjustedRowKey`?

Reply:
Yes, we should do that.